src/sgp/SGP.cc

```diff
@@ -349,13 +349,12 @@ int main(int argc, char* argv[])
 
 		#ifdef __ANDROID__
 		JNIEnv* jniEnv = (JNIEnv*)SDL_AndroidGetJNIEnv();
-		if (jniEnv == NULL) {
-			SLOGE("Failed to get jni env");
+		
+		if (setGlobalJniEnv(jniEnv) == FALSE) {
+			SLOGE("Failed to set global jni env");
```

Comment:
What would happen if we continue we `jniEnv` being null? I mean should we be throwing an exception if it is not going to work?

Reply:
Good point. Fixed now (and also when failing to find home directory).